### PR TITLE
create bake handbook direction

### PR DESCRIPTION
**Pull request by ajluk**
Issue:  #315
- add main sections titles with numbering (f.ex`H1, H2`) ,
- create Handbook Outline with title and objectives containing main section titles,
- fix headers in nested sections (the headers numbering don't correspond with sections depth in the content)  
Corresponding recipe draft PR: https://github.com/openstax/recipes/pull/69

### CODE REVIEW

File: lib/kitchen/directions/bake_handbook/v1.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    #     third_section_title = section.titles.first
+    #     third_section_title.name = 'h4'
+    #   end
+    #   page.search('> section > section > section > section').each do |section|
+    #     fourth_section_title = section.titles.first
+    #     fourth_section_title.name = 'h5'
+    #   end
+    # end
+
+    def fix_nested_section_headers(page:)
```
Comment (ajluk) [inline lib/kitchen/directions/bake_handbook/v1.rb:68]: I have rewritten this method using case statement to make the headers fir the the section depth (nesting) in the content. I left commented  how it was done previously to make the goal more clear. If someone will have a better idea how to code it, please fell free to comment! :slightly_smiling_face:

File: lib/kitchen/directions/bake_handbook/v1.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    def bake_first_section_title(page:)
+      page.search('> section').each do |section|
+        first_section_title = section.titles.first
+        first_section_title_text = first_section_title.text
+        first_section_title.replace_children(with:
+          <<~HTML
+            <span class="os-part-text">H</span>
+            <span class="os-number">#{section.count_in(:page)}</span>
+            <span class="os-divider">. </span>
+            <span class="os-text">#{first_section_title_text}</span>
```
Comment (kswanson33) [inline lib/kitchen/directions/bake_handbook/v1.rb:45]: is this variable needed? or could `first_section_title.text` be used here?

File: lib/kitchen/directions/bake_handbook/v1.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    #   end
+    #   page.search('> section > section > section > section').each do |section|
+    #     fourth_section_title = section.titles.first
+    #     fourth_section_title.name = 'h5'
+    #   end
+    # end
+
+    def fix_nested_section_headers(page:)
+      page.search('section').each do |section|
+        section_data_depth = section.raw['data-depth']
```
Comment (kswanson33) [inline lib/kitchen/directions/bake_handbook/v1.rb:70]: i believe `section.raw['data-depth']` is equivalent to `section[:'data-depth']` -- might be syntactically nicer! 🙂

**Review by kswanson33 — CHANGES_REQUESTED** [review]
This implementation contains a lot of similar searches over the content. Knowing that searching is one of the more time-intensive parts of baking, would it be possible to memoize any of the `page($.handbook).search('> section')` calls?

Otherwise, this is looking good!

**Comment by ajluk** [comment]
@kswanson33 I omitted the problem with using clipboard on objectives strings in ` bake_handbook_objectives` method we were discuss on slack by pushing them to an array and use it in the same iteration by include it in ` bake_first_section_title_and_objectives` method . It is a little different approach. Let me know what do you think about it :slightly_smiling_face:

File: lib/kitchen/directions/bake_handbook/v1.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        outline_html = <<~HTML
+          <div class="os-handbook-outline">
+            <h3 class="os-title">#{I18n.t(:handbook_outline_title)}</h3>
+          </div>
+        HTML
+        page.title.append(sibling:
+          <<~HTML
+            #{outline_html}
+          HTML
+        )
```
Comment (kswanson33) [inline lib/kitchen/directions/bake_handbook/v1.rb:26]: this can just be `page.title.append(sibling: outline_html)` -- no need for string interpolation!

**Review by kswanson33 — APPROVED** [review]
approving, but I did make one minor comment!

**Pull request merged**